### PR TITLE
feat(docs): cheatsheet

### PR DESCRIPTION
**Pull request by eladb**
Initial version of the Wing cheatsheet. Still missing some topics, but it's a great start.

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: docs/04-reference/cheatsheet.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  inflight field3: Array<str>;
+
+  init() {
+    // constructor
+    this.field1 = "hello";
+    this.field2 = 123;
+  }
+
+  setField3(value: num): num {
+    this.field3 = value;
```
Comment (tsuf239) [inline docs/04-reference/cheatsheet.md:65]: isn't it an inflight array of strings?

File: docs/04-reference/cheatsheet.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+  }
+
+  inflight init() {
```
Comment (tsuf239) [inline docs/04-reference/cheatsheet.md:68]: should it maybe be under the preflight init?

File: docs/04-reference/cheatsheet.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+* `log(s)`
+* `throw(s)`
+* `assert(condition)`
+
+# TODO
+
+* [ ] JSON support
+* [ ] enums
+* [ ] `duration`
+* [ ] `bring`
```
Comment (tsuf239) [inline docs/04-reference/cheatsheet.md:180]: Do we have a short list of all the methods available for str/num/array/map? 
I think figuring out on Google how to do `str.include `or if a language got `arr.map` is the most time-consuming thing for me when learning a new language :)

File: docs/04-reference/cheatsheet.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+  continue;
+}
+```
+
+## Classes
+
+```js
+class Foo extends Bar implements IBlink {
+  field1: str;     // <-- readonly
+  var field2: num; // <-- reassignable
```
Comment (Chriscbr) [inline docs/04-reference/cheatsheet.md:55]: style: prefer fewer keywords / new terminology
```suggestion
  field1: str;     // <-- not reassignable
  var field2: num; // <-- reassignable
```

File: docs/04-reference/cheatsheet.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+```js
+let dup = (s: str, count: num): str => {
+  // code
+};
+
+// is equivalent (unsupported YET):
+
+fn dup(s: str, counter: num): str {
+  // code
+}
```
Comment (Chriscbr) [inline docs/04-reference/cheatsheet.md:96]: I would remove this until we support it

File: docs/04-reference/cheatsheet.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+```
+
+```js
+struct Foo {
+  optValue: num?;
+}
+```
+
+```js
+fn my_func(x: str, y: num?) {
```
Comment (Chriscbr) [inline docs/04-reference/cheatsheet.md:164]:
```suggestion
fn myFunc(x: str, y: num?) {
```

File: docs/04-reference/cheatsheet.md [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+  // code
+}
+```
+
+## Utility Functions
+
+* `log(s)`
+* `throw(s)`
```
Comment (Chriscbr) [inline docs/04-reference/cheatsheet.md:172]:
```suggestion
* `log(string)`
* `throw(string)`
```

File: docs/04-reference/cheatsheet.md
Comment (Chriscbr) [inline docs/04-reference/cheatsheet.md:1]: (optional) Wing CLI commands? How do you build/run your project?

File: docs/04-reference/cheatsheet.md [hunk 8]
```diff
@@ -0,0 +1,180 @@
+# Wing for JavaScript Developers
+
+> This is under construction, let us know how we can make it better!
+
+## Primitives
+
+* `str`, `num`, `bool`
+* `Array<T>`, `MutArray<T>`
+* `Set<T>`, `MutSet<T>`
+* `Map<T>`, `MutMap<T>`
```
Comment (Chriscbr) [inline docs/04-reference/cheatsheet.md:10]: How do I initialize an array, set, or map?

File: docs/04-reference/cheatsheet.md
Comment (Chriscbr) [inline docs/04-reference/cheatsheet.md:1]: Imports / bring statements

**Comment by ShaiBer** [comment]
Keep

**Comment by eladb** [comment]
@Chriscbr I am closing this for now. Feel free to pick it up as part of docs work

**Pull request closed without merging**